src/otx/algo/detection/heads/atss_head.py

```diff
@@ -259,182 +239,17 @@ def loss_by_feat(  # type: ignore[override]
-            centerness_targets = bbox_targets.new_tensor(0.0)
-
-        # Re-weigting BG loss
-        if self.bg_loss_weight >= 0.0:
-            neg_indices = labels == self.num_classes
-            label_weights[neg_indices] = self.bg_loss_weight
-
-        if self.use_qfl:
-            labels = (labels, quality)  # For quality focal loss arg spec
-
-        # classification loss
-        loss_cls = self._get_loss_cls(cls_score, labels, label_weights, valid_label_mask, avg_factor)
-
-        return loss_cls, loss_bbox, loss_centerness, centerness_targets.sum()
-
-    def centerness_target(self, anchors: Tensor, gts: Tensor) -> Tensor:
-        """Calculate the centerness between anchors and gts.
-
-        Only calculate pos centerness targets, otherwise there may be nan.
-
-        Args:
-            anchors (Tensor): Anchors with shape (N, 4), "xyxy" format.
-            gts (Tensor): Ground truth bboxes with shape (N, 4), "xyxy" format.
-
-        Returns:
-            Tensor: Centerness between anchors and gts.
-        """
-        anchors_cx = (anchors[:, 2] + anchors[:, 0]) / 2
-        anchors_cy = (anchors[:, 3] + anchors[:, 1]) / 2
-        l_ = anchors_cx - gts[:, 0]
-        t_ = anchors_cy - gts[:, 1]
-        r_ = gts[:, 2] - anchors_cx
-        b_ = gts[:, 3] - anchors_cy
-
-        left_right = torch.stack([l_, r_], dim=1)
-        top_bottom = torch.stack([t_, b_], dim=1)
-        return torch.sqrt(
-            (left_right.min(dim=-1)[0] / left_right.max(dim=-1)[0])
-            * (top_bottom.min(dim=-1)[0] / top_bottom.max(dim=-1)[0]),
-        )
-
-    def _get_pos_inds(self, labels: Tensor) -> Tensor:
-        bg_class_ind = self.num_classes
-        return ((labels >= 0) & (labels < bg_class_ind)).nonzero().squeeze(1)
-
-    def _get_loss_cls(
-        self,
-        cls_score: Tensor,
-        labels: Tensor,
-        label_weights: Tensor,
-        valid_label_mask: Tensor,
-        avg_factor: Tensor,
-    ) -> Tensor:
-        if isinstance(self.loss_cls, CrossSigmoidFocalLoss):
-            loss_cls = self.loss_cls(
-                cls_score,
-                labels,
-                label_weights,
-                avg_factor=avg_factor,
-                valid_label_mask=valid_label_mask,
-            )
-        else:
-            loss_cls = self.loss_cls(cls_score, labels, label_weights, avg_factor=avg_factor)
-        return loss_cls
-
-    def _get_loss_centerness(
-        self,
-        avg_factor: Tensor,
-        pos_centerness: Tensor,
-        centerness_targets: Tensor,
-    ) -> Tensor:
-        return self.loss_centerness(pos_centerness, centerness_targets, avg_factor=avg_factor)
-
-    def _get_loss_bbox(
-        self,
-        pos_bbox_targets: Tensor,
-        pos_bbox_pred: Tensor,
-        centerness_targets: Tensor,
-    ) -> Tensor:
-        return self.loss_bbox(pos_bbox_pred, pos_bbox_targets, weight=centerness_targets, avg_factor=1.0)
+        return {
```

Comment:
It can be confusing that function name is `loss_by_feat` but it doesn't return loss. How about changing name?

Reply:
I think it'd better to compare it with `predict_by_feat` and `export_by_feat`, and combine redundant parts if there are.
But it can be a huge work, I'm thinking about splitting PRs or not.